mm2src/coins/utxo/utxo_common.rs

```diff
@@ -1230,6 +1387,66 @@ pub fn send_taker_spends_maker_payment<T: UtxoCommonOps + SwapOps>(
     Box::new(fut.boxed().compat())
 }
 
+pub fn send_watcher_spends_maker_payment<T: UtxoCommonOps + SwapOps>(
+    coin: T,
+    payment_input: WatcherSpendsMakerPaymentInput<'_>,
+) -> TransactionFut {
+    let taker_address = try_tx_fus!(address_from_raw_pubkey(
+        payment_input.taker_pub,
+        coin.as_ref().conf.pub_addr_prefix,
+        coin.as_ref().conf.pub_t_addr_prefix,
+        coin.as_ref().conf.checksum_type,
+        coin.as_ref().conf.bech32_hrp.clone(),
+        coin.addr_format().clone(),
+    ));
+
+    let mut prev_transaction: UtxoTx =
+        try_tx_fus!(deserialize(payment_input.maker_payment_tx).map_err(|e| ERRL!("{:?}", e)));
+    prev_transaction.tx_hash_algo = coin.as_ref().tx_hash_algo;
+
+    let script_data = Builder::default()
+        .push_data(payment_input.secret)
+        .push_opcode(Opcode::OP_0)
+        .into_script();
+
+    let redeem_script = payment_script(
+        payment_input.time_lock,
+        &*dhash160(payment_input.secret),
+        &try_tx_fus!(Public::from_slice(payment_input.maker_pub)),
+        &try_tx_fus!(Public::from_slice(payment_input.taker_pub)),
+    )
+    .into();
+
+    let signature: Bytes = payment_input.signature.into();
+    let lock_time = payment_input.time_lock;
+
+    let fut = async move {
+        let fee = try_tx_s!(coin.get_htlc_spend_fee(DEFAULT_SWAP_TX_SPEND_SIZE).await);
+        let script_pubkey = output_script(&taker_address, ScriptType::P2PKH).to_bytes();
+        let output = TransactionOutput {
+            value: prev_transaction.outputs[0].value - fee,
+            script_pubkey,
+        };
```

Comment:
Sorry just a personal question..

Is there a guarantee that `prev_transaction.outputs[0].value` will always be greater than `fee` ?.. This was the cause of an `attempt to subtract with overflow` panic I encountered in this issue #1304 while performing a `KMD` swap sometimes ago...even tho I only came across it multiple times, I have been unable to reproduce ever since